dftimewolf/lib/collectors/grr_hosts.py

```diff
@@ -316,52 +317,62 @@ def setup(self,
     self._clients = self.find_clients(hosts)
     self.use_tsk = use_tsk
 
+  def _process_thread(self, client):
+    """Process a single GRR client.
+
+    Args:
+      client: a GRR client object.
+    """
+    system_type = client.data.os_info.system
+    fqdn = client.data.os_info.fqdn.lower()
+    client_dir = os.path.join(self.output_path)
+    if not os.path.isdir(client_dir):
+      os.makedirs(client_dir)
+    print('System type: {0:s}'.format(system_type))
+
+    # If the list is supplied by the user via a flag, honor that.
+    artifact_list = []
+    if self.artifacts:
+      print('Artifacts to be collected: {0:s}'.format(self.artifacts))
+      artifact_list = self.artifacts
+    else:
+      default_artifacts = self.artifact_registry.get(system_type, None)
```

Comment:
Handle KeyError here instead of defaulting to None, there's no point trying to collect the "None" artifact, and the print() below will raise in Python 3.

Reply:
I'm keeping the `None` here and adding a condition to the statements below.